client/src/modules/unsubscribe/index.tsx

```diff
@@ -45,13 +45,18 @@ const UnsubscribePage: NextPage = () => {
         <Heading>Unsubscribing</Heading>
         {!loading && !data && !error ? (
           <form onSubmit={handleSubmit(onUnsubscribe)}>
-            <HStack>
-              <Checkbox
-                {...register('confirm' as const, { required: true })}
-                value="confirm"
-              >
-                Confirm unsubscribing
-              </Checkbox>
+            <HStack marginTop="1em">
+              <fieldset>
+                <Text as="legend" srOnly>
+                  interacting with {data}&apos;s subscription
+                </Text>
+                <Checkbox
+                  {...register('confirm' as const, { required: true })}
+                  value="confirm"
+                >
+                  Confirm unsubscribing
+                </Checkbox>
+              </fieldset>
```

Comment:
Maybe it's better if this is a button with the text unsubscribe, with a paragraph telling what is actually happening when you are unsubscribed  🤔